aladdin-container.sh

```diff
@@ -118,6 +118,35 @@ function environment_init() {
 
 }
 
+function add_and_set_authentication_users() {
+    if $AUTHENTICATION_ENABLED; then
+        jq -r '.|keys[]' <<< "$AUTHENTICATION_ROLES" | while read name ; do
+            role_arn="$(jq -r --arg name "$name" '.[$name]' <<< $AUTHENTICATION_ROLES)"
+            _add_authentication_user_to_kubeconfig "$name" "$role_arn"
+        done
+        kubectl config set-context "$NAMESPACE.$CLUSTER_NAME" --cluster "$CLUSTER_NAME" --namespace="$NAMESPACE" --user "$AUTHENTICATION_ALADDIN_ROLE"
+    fi
+}
+
+function _add_authentication_user_to_kubeconfig() {
+    name="$1"
+    role_arn="$2"
+cat <<EOT >> $HOME/.kube/config
+- name: $1
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1alpha1
+      args:
+      - token
+      - -i
+      - $CLUSTER_NAME
+      - -r
+      - $role_arn
+      command: aws-iam-authenticator
+EOT
```

Comment:
Couldn't figure out how to leave this indented without breaking syntax

Reply:
This kubeconfig file could alternatively just be in s3 and we could pull it down, rather than generated on the fly like it is here

Reply:
I think generation inline is fine.

Reply:
There's a way to do it but it involves using tabs for indentation which is likely to create issues in the future as I know I've run into code that works that way and not known I was breaking it by using spaces :\  ^EOT seems like a fine compromise to me.

Reply:
You *could* indent just the initial cat command ...